contrib/babelfishpg_tsql/sql/object_definition_tsql.sql

```diff
@@ -11,4 +11,51 @@ CREATE TABLE sys.babelfish_view_def (
 );
 GRANT SELECT ON sys.babelfish_view_def TO PUBLIC;
 
-SELECT pg_catalog.pg_extension_config_dump('sys.babelfish_view_def', '');
\ No newline at end of file
+-- BBF_PARTITION_FUNCTION
+-- This catalog stores the metadata of partition funtions.
+CREATE TABLE sys.babelfish_partition_function
+(
+  dbid SMALLINT NOT NULL,
+  function_id INT NOT NULL UNIQUE,
+  partition_function_name sys.NVARCHAR(128) NOT NULL COLLATE sys.database_default,
+  input_type sys.sysname,
+  partition_option sys.bit,
+  range_values sys.sql_variant[],
```

Comment:
IIUC, there could be at max 15000 partitions, can we have some check constraint here to ensure the same?

Reply:
We are already checking the partition limit before inserting into catalog so check will be redundant here but yes to avoid user direct manipulation of these catalogs, I will add that check here also.